forecasttools/recode_locations.py

```diff
@@ -14,31 +14,159 @@ def loc_abbr_to_flusight_code(
+    it to FluSight location codes using
+    location_table, which is a Polars
+    dataframe contained in forecasttools.
 
     Parameters
     ----------
     df
         A Polars dataframe with a location
-        column.
+        column consisting of US
+        jurisdictional abbreviations.
     location_col
         The name of the dataframe's location
         column.
 
     Returns
     -------
     pl.DataFrame
-        A recoded locations dataframe.
+        A Polars dataframe with the location
+        column formatted as FluSight location
+        codes.
     """
     # get location table
     loc_table = forecasttools.location_table
-    # recode and replaced existing loc abbrs with loc codes
+    # recode and replaced existing loc abbrs
+    # with loc codes
     loc_recoded_df = df.with_columns(
         location=pl.col("location").replace(
             old=loc_table["short_name"],
             new=loc_table["location_code"],
         )
     )
     return loc_recoded_df
+
+
+def loc_flusight_code_to_abbr(
+    df: pl.DataFrame, location_col: str
+) -> pl.DataFrame:
+    """
+    Takes the location columns of a Polars
+    dataframe (formatted as FluSight codes for
+    US two-letter jurisdictions) and recodes
+    it to US jurisdictional abbreviations,
+    using location_table, which is a Polars
+    dataframe contained in forecasttools.
+
+    Parameters
+    ----------
+    df
+        A Polars dataframe with a location
+        column consisting of US
+        jurisdictional FluSight codes.
+    location_col
+        The name of the dataframe's location
+        column.
+
+    Returns
+    -------
+    pl.DataFrame
+        A Polars dataframe with the location
+        column formatted as US two-letter
+        jurisdictional abbreviations.
+    """
+    # get forecasttools location table
+    loc_table = forecasttools.location_table
+    # recode location codes to location abbreviations
+    loc_recoded_df = df.with_columns(
+        pl.col(location_col).replace(
+            old=loc_table["location_code"], new=loc_table["short_name"]
+        )
+    )
+    return loc_recoded_df
+
+
+def to_location_table_column(location_format: str) -> str:
+    """
+    Maps a location format string to the
+    corresponding column name in the hubserve
+    location table. For example, "flusight"
+    maps to "location_code" in forecasttool's
+    location_table.
+
+    Parameters
+    ----------
+    location_format
+        The format string ("abbr",
+        "flusight", or "long_name").
+
+    Returns
+    -------
+    str
+        Returns the corresponding column name
+        from the location table.
+    """
+    col_dict = {
+        "abbr": "short_name",
+        "flusight": "location_code",
+        "long_name": "long_name",
+    }
+    col = col_dict.get(location_format)
+    if col is None:
+        raise KeyError(
+            f"Unknown location format {location_format}. Expected 'abbr', 'flusight', or 'long_name'."
```

Comment:
pull the names of the valid formats from the dictionary keys?

Reply:
Not sure you are questioning here.

If the name is not a valid (supported) format, then the name will not be in the dictionary keys of `col_dict` and an error will be thrown.

Reply:
The error message says `Expected 'abbr', 'flusight', or 'long_name'."`. Those are just the keys from `col_dict`. You should use that fact to make the error message more automated. Then, it won't be out of date if you ever add new keys to `col_dict`.

Reply:
I see. Nice observation.